--- a/objectives.cpp
+++ b/objectives.cpp
@@ -439,7 +439,7 @@
 	if ( status != ObjectiveManager::STATUS_HIDDEN ) {
 		WideStringClass message;
 		WideStringClass description = TranslateDBClass::Get_String( short_description_id );
-		message.Format( TRANSLATE (IDS_OBJ_NEW_OBJ), objective->Type_To_Name(), description );
+		message.Format( TRANSLATE (IDS_OBJ_NEW_OBJ), objective->Type_To_Name(), (const WCHAR *)description );
 		CombatManager::Get_Message_Window ()->Add_Message (message, objective->Type_To_Base_Color());
 
 		HUDClass::Add_Objective( type );
@@ -517,7 +517,7 @@
 		WideStringClass message;
 		if (is_unhiding && (status != ObjectiveManager::STATUS_ACCOMPLISHED)) {
 			WideStringClass description = TranslateDBClass::Get_String( objective->ShortDescriptionID );
-			message.Format( TRANSLATE (IDS_OBJ_NEW_OBJ), objective->Type_To_Name(), description );
+			message.Format( TRANSLATE (IDS_OBJ_NEW_OBJ), objective->Type_To_Name(), (const WCHAR *)description );
 
 			HUDClass::Add_Objective( objective->Type );
 		} else {
