--- a/mapmgr.cpp
+++ b/mapmgr.cpp
@@ -93,7 +93,7 @@
 	//	Strip off the path if necessary
 	//
 	filename = MapTextureName;
-	char *dir_delimiter			= ::strrchr (MapTextureName, '\\');
+	const char *dir_delimiter		= ::strrchr (MapTextureName, '\\');
 	if (dir_delimiter != NULL) {
 		filename = (dir_delimiter + 1);
 	}
@@ -114,7 +114,7 @@
 	//	Strip off the path if necessary
 	//
 	StringClass filename_only	= filename;
-	char *dir_delimiter			= ::strrchr (filename, '\\');
+	const char *dir_delimiter		= ::strrchr (filename, '\\');
 	if (dir_delimiter != NULL) {
 		filename_only = (dir_delimiter + 1);
 	}
