--- a/campaign.cpp
+++ b/campaign.cpp
@@ -495,13 +495,16 @@
 {
 	// Find Backdrop Index
 	BackdropIndex = 0;
+	bool found = false;
 	for ( int i = 0; i < BackdropDescriptions.Count(); i++ ) {
 		if ( BackdropDescriptions[i].State == state_number ) {
 			BackdropIndex = i;
+			found = true;
+			break;
 		}
 	}
 
-	if ( BackdropIndex == 0 ) {
+	if ( !found ) {
 		Debug_Say(( "Failed to find load menu for state %d\n", state_number ));
 	}
 }
