--- a/AnnounceEvent.cpp
+++ b/AnnounceEvent.cpp
@@ -39,13 +39,15 @@
 #include "apppackettypes.h"
 #include "networkobjectmgr.h"
 #include "cnetwork.h"
+#include "playermanager.h"
+#include "player.h"
 #include "gamemode.h"
 #include "playertype.h"
 #include "translateobj.h"
 #include "translatedb.h"
 #include "wwaudio.h"
 #include "messagewindow.h"
-#include <wwlib\widestring.h>
+#include "widestring.h"
 #include "cncmodesettings.h"
 #include "floodprotectionmgr.h"
 
@@ -333,7 +335,7 @@
 	packet.Get(mToID);
 	packet.Get(mFromID);
 	packet.Get(mAnnouncementID);
-	packet.Add(mRadioCmdID);
+	packet.Get(mRadioCmdID);
 
 	BYTE type = 0;
 	packet.Get(type);
--- a/floodprotectionmgr.cpp
+++ b/floodprotectionmgr.cpp
@@ -39,6 +39,7 @@
 #include "messagewindow.h"
 #include "translatedb.h"
 #include "string_ids.h"
+#include "systimer.h"
 
 
 //////////////////////////////////////////////////////////////////////
@@ -70,7 +71,7 @@
 {
 	const int DECAY_TIME	= 15000;
 
-	uint32 curr_time = ::GetTickCount ();
+	uint32 curr_time = TIMEGETTIME();
 
 	//
 	//	Loop over all the entries in the list
@@ -124,7 +125,7 @@
 		//	Add a new entry to the list
 		//
 		FLOOD_ENTRY entry;
-		entry.time		= ::GetTickCount ();
+		entry.time		= TIMEGETTIME();
 		entry.text_len	= 0;
 		FloodList.Add (entry);
 	}
