--- a/datasafe.h
+++ b/datasafe.h
@@ -828,12 +828,21 @@
 **
 **
 */
+#if defined(_MSC_VER)
 #define DECLARE_DATA_SAFE(type) 																\
 	DataSafeClass<type> DataSafe##type;														\
 	int DataSafeClass<type>::Type;															\
 	char DataSafeClass<type>::ReturnList[MAX_OBJECT_COPIES][sizeof(type)];		\
 	int DataSafeClass<type>::ReturnIndex;													\
 	int DataSafeClass<type>::MinSlop;
+#else
+#define DECLARE_DATA_SAFE(type) 																\
+	DataSafeClass<type> DataSafe##type;														\
+	template <> int DataSafeClass<type>::Type = 0;								\
+	template <> char DataSafeClass<type>::ReturnList[MAX_OBJECT_COPIES][sizeof(type)] = {}; \
+	template <> int DataSafeClass<type>::ReturnIndex = 0;						\
+	template <> int DataSafeClass<type>::MinSlop = 0;
+#endif
 
 
 
