--- a/dlgcncteaminfo.cpp
+++ b/dlgcncteaminfo.cpp
@@ -53,8 +53,6 @@
 #include "vehicle.h"
 #include "assets.h"
 #include "translatedb.h"
-#include "WOLGMode.h"
-#include <WWOnline\WOLUser.h>
 #include "string_ids.h"
 #include "mousemgr.h"
 #include "directinput.h"
--- a/dlgcncbattleinfo.cpp
+++ b/dlgcncbattleinfo.cpp
@@ -52,8 +52,6 @@
 #include "damage.h"
 #include "vehicle.h"
 #include "resource.h"
-#include "WOLGMode.h"
-#include <WWOnline\WOLUser.h>
 #include "translatedb.h"
 #include "string_ids.h"
 #include "mousemgr.h"
@@ -74,5 +72,5 @@
-static const BUILDING_COUNT		= 5;
-static const BUILDING_SLOT_COUNT	= 6;
+static const int BUILDING_COUNT		= 5;
+static const int BUILDING_SLOT_COUNT	= 6;
 
 
 ////////////////////////////////////////////////////////////////
--- a/dlgcncserverinfo.cpp
+++ b/dlgcncserverinfo.cpp
@@ -53,8 +53,6 @@
 #include "damage.h"
 #include "vehicle.h"
 #include "resource.h"
-#include "WOLGMode.h"
-#include <WWOnline\WOLUser.h>
 #include "translatedb.h"
 #include "string_ids.h"
 #include "mousemgr.h"
