--- a/Test_Cinematic.cpp
+++ b/Test_Cinematic.cpp
@@ -208,14 +208,11 @@
 	*/
 	void	Load_Control_File( const char * filename ) 
 	{
-		Commands->Debug_Message( "Loading Control File %s\n", (int)filename );
-
-		char full_filename[80];
-		sprintf( full_filename, "DATA\\%s", filename );
-//		FILE * in = fopen( full_filename, "rt" );
+		Commands->Debug_Message( "Loading Control File %s\n", filename );
+
 		int handle = Commands->Text_File_Open( filename );
 		if ( handle == 0 ) {
-			Commands->Debug_Message( "Failed to open %s\n", (int)full_filename );
+			Commands->Debug_Message( "Failed to open DATA\\%s\n", filename );
 			return;
 		}
