--- a/Test_RMV_Toolkit.cpp
+++ b/Test_RMV_Toolkit.cpp
@@ -191,9 +191,7 @@
 		{
 			busy = true;
 			terminal_id = Commands->Get_ID(sender);
-			char *anim;
-			anim = (char *)param;
-			int_anim = (int)anim;
+			int_anim = param;
 			ActionParamsStruct params;
 			params.Set_Basic(this, 70, TECHNICIAN_MOVEMENT);
 			params.Set_Movement(sender, emergency ? RUN : WALK, 0.75f);
@@ -214,9 +212,7 @@
				float facing = Commands->Get_Facing(terminal);
				Commands->Set_Facing(obj, facing + 180);
 			}
-			char *anim;
-			anim = (char *)int_anim;
-			anim = "s_a_human.h_a_con2";
+			char *anim = "s_a_human.h_a_con2";
 			ActionParamsStruct params;
 			params.Set_Basic(this, 70, TECHNICIAN_ANIMATION);
 			params.Set_Animation(anim, false);
