--- a/animobj.cpp
+++ b/animobj.cpp
@@ -764,7 +764,3 @@
 			if ( ModeAnim.Motion->Has_Embedded_Sounds() ) {
-#if defined(RENEGADE_VITA_PORT)
-				ModeAnim.PrevFrame = ModeAnim.Frame;
-#else
 				ModeAnim.PrevFrame = AnimatedSoundMgrClass::Trigger_Sound(ModeAnim.Motion, ModeAnim.PrevFrame, ModeAnim.Frame, Get_Transform ());
-#endif
 			}
@@ -780,15 +776,7 @@
 			if ( ModeInterp.Motion0->Has_Embedded_Sounds() ) {
-#if defined(RENEGADE_VITA_PORT)
-				ModeInterp.PrevFrame0 = ModeInterp.Frame0;
-#else
 				ModeInterp.PrevFrame0 = AnimatedSoundMgrClass::Trigger_Sound(ModeInterp.Motion0, ModeInterp.PrevFrame0, ModeInterp.Frame0, Get_Transform ());
-#endif
 			}
 
 			if ( ModeInterp.Motion1->Has_Embedded_Sounds() ) {
-#if defined(RENEGADE_VITA_PORT)
-				ModeInterp.PrevFrame1 = ModeInterp.Frame1;
-#else
 				ModeInterp.PrevFrame1 = AnimatedSoundMgrClass::Trigger_Sound(ModeInterp.Motion1, ModeInterp.PrevFrame1, ModeInterp.Frame1, Get_Transform ());
-#endif
 			}
@@ -809,9 +797,5 @@
 				if ( motion != NULL && motion->Has_Embedded_Sounds() ) {
-#if defined(RENEGADE_VITA_PORT)
-					ModeCombo.AnimCombo->Set_Prev_Frame(index, ModeCombo.AnimCombo->Get_Frame(index));
-#else
 					float prev_frame = AnimatedSoundMgrClass::Trigger_Sound(motion, ModeCombo.AnimCombo->Get_Prev_Frame(index),
 															ModeCombo.AnimCombo->Get_Frame(index), Get_Transform ());
 					ModeCombo.AnimCombo->Set_Prev_Frame(index, prev_frame);
-#endif
 				}
