--- a/AudibleSound.h
+++ b/AudibleSound.h
@@ -74,7 +74,7 @@
 //
 //	Typedefs
 //
-typedef unsigned long MILES_HANDLE;
+typedef uintptr_t MILES_HANDLE;
 
 typedef enum
 {
--- a/soundhandle.h
+++ b/soundhandle.h
@@ -90,7 +90,7 @@
 	//
 	//	Initialization
 	//	
-	virtual void	Set_Miles_Handle (uint32 handle) = 0;
+	virtual void	Set_Miles_Handle (uintptr_t handle) = 0;
 	virtual void	Initialize (SoundBufferClass *buffer);
 
 	//
--- a/sound2dhandle.h
+++ b/sound2dhandle.h
@@ -76,7 +76,7 @@
 	//
 	//	Inherited
 	//
-	void							Set_Miles_Handle (uint32 handle);
+	void							Set_Miles_Handle (uintptr_t handle);
 	void							Initialize (SoundBufferClass *buffer);
 	void							Start_Sample (void);
 	void							Stop_Sample (void);
--- a/sound3dhandle.h
+++ b/sound3dhandle.h
@@ -77,7 +77,7 @@
 	//
 	//	Inherited
 	//
-	void							Set_Miles_Handle (uint32 handle);
+	void							Set_Miles_Handle (uintptr_t handle);
 	void							Initialize (SoundBufferClass *buffer);
 	void							Start_Sample (void);
 	void							Stop_Sample (void);
--- a/soundstreamhandle.h
+++ b/soundstreamhandle.h
@@ -78,7 +78,7 @@
 	//
 	//	Inherited
 	//
-	void							Set_Miles_Handle (uint32 handle);
+	void							Set_Miles_Handle (uintptr_t handle);
 	void							Initialize (SoundBufferClass *buffer);
 	void							Start_Sample (void);
 	void							Stop_Sample (void);
--- a/sound2dhandle.cpp
+++ b/sound2dhandle.cpp
@@ -365,7 +365,7 @@
 //
 //////////////////////////////////////////////////////////////////////
 void
-Sound2DHandleClass::Set_Miles_Handle (uint32 handle)
+Sound2DHandleClass::Set_Miles_Handle (uintptr_t handle)
 {
 	SampleHandle = (HSAMPLE)handle;
 	return ;
--- a/sound3dhandle.cpp
+++ b/sound3dhandle.cpp
@@ -377,7 +377,7 @@
 //
 //////////////////////////////////////////////////////////////////////
 void
-Sound3DHandleClass::Set_Miles_Handle (uint32 handle)
+Sound3DHandleClass::Set_Miles_Handle (uintptr_t handle)
 {
 	WWASSERT (SampleHandle == (H3DSAMPLE)INVALID_MILES_HANDLE);
 
--- a/soundstreamhandle.cpp
+++ b/soundstreamhandle.cpp
@@ -359,7 +359,7 @@
 //
 //////////////////////////////////////////////////////////////////////
 void
-SoundStreamHandleClass::Set_Miles_Handle (uint32 handle)
+SoundStreamHandleClass::Set_Miles_Handle (uintptr_t handle)
 {
 	SampleHandle = (HSAMPLE)handle;
 	return ;
